vm/src/builtins/list.rs

```diff
@@ -471,19 +479,61 @@ impl PyValue for PyListIterator {
 impl PyListIterator {
     #[pymethod(name = "__length_hint__")]
     fn length_hint(&self) -> usize {
-        let list = self.list.borrow_vec();
-        let pos = self.position.load();
-        list.len().saturating_sub(pos)
+        match self.status.load() {
+            Active => {
+                let list = self.list.borrow_vec();
+                let pos = self.position.load();
+                list.len().saturating_sub(pos)
+            }
+            Exhausted => 0,
+        }
+    }
+
+    #[pymethod(name = "__setstate__")]
+    fn setstate(&self, state: PyObjectRef, vm: &VirtualMachine) -> PyResult {
+        // When we're exhausted, just return.
+        if let Exhausted = self.status.load() {
+            return Ok(vm.ctx.none());
+        }
+        if let Some(i) = state.payload::<PyInt>() {
+            let position = std::cmp::min(
+                int::try_to_primitive(i.as_bigint(), vm).unwrap_or(0),
+                self.list.len(),
+            );
+            self.position.store(position);
+            Ok(vm.ctx.none())
+        } else {
+            Err(vm.new_type_error("an integer is required.".to_owned()))
+        }
+    }
```

Comment:
this function looks like always returning `Ok(vm.vtx.none())` when it returns `Ok`. By using `PyResult<()>`, `Ok(())` will work instead of returning `vm.ctx.none()` everytime.